src/gui/VerificationValidationWidget.cpp

```diff
@@ -443,6 +488,490 @@ void VerificationValidationWidget::testListSelection(QListWidgetItem* test_click
+        showNewTestDialog();
+        return;
+    }
+    QString testCmd = testCmdInput->text();
+
+    QString testCategory = "";
+    if(!testCategoryInput->text().simplified().isEmpty()){
+        testCategory = testCategoryInput->text();
+    }
+
+    bool hasVar = false;
+    for(int i = 0; i < isVarList.size(); i++){
+        if(isVarList[i]->checkState()){
+            hasVar = true;
+            break;
+        } else {
+            if(!varInputList[i]->text().simplified().isEmpty()){
+                hasVar = true;
+                break;
+            }
+        }
+    }
+
+    QSqlQuery* q = new QSqlQuery(getDatabase());
+    q->prepare("INSERT INTO Tests (testName, testCommand, hasValArgs, category) VALUES (:testName, :testCommand, :hasValArgs, :category)");
+    q->bindValue(":testName", testName);
+    q->bindValue(":testCommand", testCmd);
+    q->bindValue(":hasValArgs", hasVar);
+    q->bindValue(":category", testCategory);
+    q->exec();
+
+    QString testID = q->lastInsertId().toString();
+    for(int i = 0; i < addToSuiteList->count(); i++){
+        QListWidgetItem* item = addToSuiteList->item(i);
+        if(item->checkState()){
+            q->prepare("SELECT id FROM TestSuites WHERE suiteName = ?");
+            q->addBindValue(item->text());
+            q->exec();
+            int suiteID = -1;
+            while(q->next()){
+                suiteID = q->value(0).toInt();
+            }
+            q->prepare("INSERT INTO TestsInSuite (testSuiteID, testID) VALUES (:suiteID, :testID)");
+            q->bindValue(":suiteID", suiteID);
+            q->bindValue(":testID", testID);
+            q->exec();
+        }
+    }
+
+    int argIdx = 1;
+    for(int i = 0; i < argInputList.size(); i++){
+        if(argInputList[i]->text().simplified().isEmpty() && !isVarList[i]->checkState() && varInputList[i]->text().simplified().isEmpty()){
+            continue;
+        }
+        bool isVariable = false;
+        if(isVarList[i]->checkState() || !varInputList[i]->text().simplified().isEmpty()){
+            isVariable = true;
+        }
+        q->prepare("INSERT INTO TestArg (testID, argIdx, arg, isVarArg, defaultVal) VALUES (:testID, :argIdx, :arg, :isVarArg, :defaultVal)");
+        q->bindValue(":testID", testID);
+        q->bindValue(":argIdx", argIdx);
+        q->bindValue(":arg", argInputList[i]->text());
+        q->bindValue(":isVarArg", isVariable);
+        q->bindValue(":defaultVal", varInputList[i]->text());
+        dbExec(q);
+        argIdx += 1;
+    }
+
+    setupUI();
+}
+
+void VerificationValidationWidget::addArgForm() {
+    int n = argInputList.size()+1;
+    QString boxTitle = "Argument Input %1";
+    QGroupBox* argField = new QGroupBox(boxTitle.arg(n));
+    QFormLayout* argForm = new QFormLayout();
+    QLineEdit* argInput = new QLineEdit();
+    argForm->addRow("Argument: ", argInput);
+    QCheckBox* isVar = new QCheckBox();
+    argForm->addRow("Has variable: ", isVar);

```

Comment:
able to infer whether argument has a variable based on if variable textbox is empty or not